wear/src/main/java/io/homeassistant/companion/android/tiles/ShortcutsTile.kt

```diff
@@ -0,0 +1,218 @@
+                    ).build()
+                ).build()
+        }
+
+    override fun onResourcesRequest(requestParams: ResourcesRequest): ListenableFuture<Resources> =
+        serviceScope.future {
+            val density = requestParams.deviceParameters!!.screenDensity
+            val iconSizePx = (ICON_SIZE * density).roundToInt()
+            val entities = getEntities()
+
+            Resources.Builder()
+                .setVersion(entities.toString())
+                .apply {
+                    entities.map { entity ->
+                        val entityId = entity.split(",")[0]
+                        val entityIcon = entity.split(",")[2]
+
+                        // Find icon name
+                        val iconName: String = if (entityIcon.startsWith("mdi")) {
+                            entityIcon.split(":")[1]
+                        } else {
+                            "palette" // Default scene icon
+                        }
+
+                        // Create Bitmap from icon name
+                        val iconBitmap = IconicsDrawable(this@ShortcutsTile, "cmd-$iconName").apply {
+                            colorInt = Color.WHITE
+                            sizeDp = ICON_SIZE.roundToInt()
+                            backgroundColor = IconicsColor.colorRes(R.color.colorOverlay)
+                        }.toBitmap(iconSizePx, iconSizePx, Bitmap.Config.RGB_565)
+
+                        // Make array of bitmap
+                        val bitmapData = ByteBuffer.allocate(iconBitmap.byteCount).apply {
+                            iconBitmap.copyPixelsToBuffer(this)
+                        }.array()
+
+                        // link the entity id to the bitmap data array
+                        entityId to ResourceBuilders.ImageResource.Builder()
+                            .setInlineResource(
+                                ResourceBuilders.InlineImageResource.Builder()
+                                    .setData(bitmapData)
+                                    .setWidthPx(iconSizePx)
+                                    .setHeightPx(iconSizePx)
+                                    .setFormat(ResourceBuilders.IMAGE_FORMAT_RGB_565)
+                                    .build()
+                            )
+                            .build()
+                    }.forEach { (id, imageResource) ->
+                        addIdToImageMapping(id, imageResource)
+                    }
+                }
+                .build()
+        }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        // Cleans up the coroutine
+        serviceJob.cancel()
+    }
+
+    private suspend fun getEntities(): List<String> {
+        DaggerTilesComponent.builder()
+            .appComponent((applicationContext as GraphComponentAccessor).appComponent)
+            .build()
+            .inject(this@ShortcutsTile)
+
+        return integrationUseCase.getTileShortcuts()
+    }
```

Comment:
As mentioned previously, this should be an object with properties rather than just a string.